packages/tools/src/tools/SegmentationIntersectionTool.ts

```diff
@@ -0,0 +1,283 @@
+   */
+  renderAnnotation = (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ): boolean => {
+    const { viewport, FrameOfReferenceUID } = enabledElement;
+
+    let renderStatus = false;
+
+    const annotations = getAnnotations(this.getToolName(), FrameOfReferenceUID);
+    if (!annotations?.length) {
+      return renderStatus;
+    }
+    const annotation = annotations[0];
+    const { annotationUID } = annotation;
+    const actorsWorldPointsMap = annotation.data.actorsWorldPointsMap;
+
+    calculateSurfaceSegmentationIntersectionsForViewport(
+      actorsWorldPointsMap,
+      viewport
+    );
+    const actorEntries = viewport.getActors();
+    const { focalPoint } = viewport.getCamera();
+    const focalPointString = pointToString(focalPoint);
+    actorEntries.forEach((actorEntry) => {
+      if (!actorEntry?.clippingFilter) {
+        return;
+      }
+      const actorWorldPointMap = actorsWorldPointsMap.get(actorEntry.uid);
+      if (!actorWorldPointMap) {
+        return;
+      }
+      if (!actorWorldPointMap.get(focalPointString)) {
+        return;
+      }
+      let polyLineIdx = 1;
+      const { worldPointsSet, color } =
+        actorWorldPointMap.get(focalPointString);
+      worldPointsSet.forEach((worldPoints) => {
+        const canvasPoints = worldPoints.map((point) =>
+          viewport.worldToCanvas(point)
+        );
+
+        const options = {
+          color: 'none',
+          fillColor: color,
+          fillOpacity: 0.5,
+          connectLastToFirst: true,
+        };
+
+        const polyLineUID = actorEntry.uid + '#' + polyLineIdx;
+        drawPolyline(
+          svgDrawingHelper,
+          annotationUID,
+          polyLineUID,
+          canvasPoints,
+          options
+        );
+        polyLineIdx++;
+      });
+    });
+
+    renderStatus = true;
+    return renderStatus;
+  };
+}
+
+/**
+ * Calculates surface intersections points for all surface actors in a viewport
+ * generating a set of polyline points for each actor
+ * @param actorWorldPointsMap
+ * @param viewport
+ */
+function calculateSurfaceSegmentationIntersectionsForViewport(
+  actorsWorldPointsMap,
+  viewport
+) {
+  const actorEntries = viewport.getActors();
+  const { focalPoint } = viewport.getCamera();
+  const focalPointString = pointToString(focalPoint);
+  actorEntries.forEach((actorEntry) => {
+    if (actorEntry?.clippingFilter) {
+      let actorWorldPointsMap = actorsWorldPointsMap.get(actorEntry.uid);
+      if (!actorWorldPointsMap) {
+        actorWorldPointsMap = new Map();
+        actorsWorldPointsMap.set(actorEntry.uid, actorWorldPointsMap);
+      }
+      if (!actorWorldPointsMap.get(focalPointString)) {
+        const polyData = actorEntry.clippingFilter.getOutputData();
+        let worldPointsSet = getPolyDataPoints(polyData);
+        if (worldPointsSet) {
+          worldPointsSet = removeExtraPoints(viewport, worldPointsSet);
```

Comment:
I commented this line out (to not remove extra points) and everything worked fine. Do you have a reason to include this?